crypto/bn/bn_gcd.c

```diff
@@ -531,7 +531,11 @@ BIGNUM *BN_mod_inverse(BIGNUM *in,
     rv = int_bn_mod_inverse(in, a, n, ctx, &noinv);
     if (noinv)
         ERR_raise(ERR_LIB_BN, BN_R_NO_INVERSE);
-    BN_CTX_free(new_ctx);
+
+    if (new_ctx != NULL) {
+        BN_CTX_free(new_ctx);
+        ctx = NULL;
+    }
```

Comment:
This change does not make any sense. These variables are going out of scope after the return below.